Scripts/Runtime/Data/Collections/UnsafeTypedStream.cs

```diff
@@ -674,6 +691,21 @@ private void CheckForEndOfBlock()
                 return ref UnsafeUtility.AsRef<T>(readPointer);
             }
 
+            internal void ReadBlock(byte* dstStart, int dstStartIndex, int elementsToRead)
+            {
+#if ENABLE_UNITY_COLLECTIONS_CHECKS
+                Assert.IsTrue(Count > 0 && m_CurrentReadBlock != null && m_ReaderHead != null);
```

Comment:
Should we add a condition to make sure m_ReaderHead + elementsToRead stays within the bounds of the lane?

Reply:
Yeah that's a good idea. Done.